cmd/soroban-rpc/internal/ledgerentry_storage/db.go

```diff
@@ -0,0 +1,295 @@
+package ledgerentry_storage
+
+import (
+	"context"
+	"database/sql"
+	"embed"
+	"fmt"
+	"strconv"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/jmoiron/sqlx"
+	_ "github.com/mattn/go-sqlite3"
+	migrate "github.com/rubenv/sql-migrate"
+	"github.com/stellar/go/support/errors"
+	"github.com/stellar/go/xdr"
+)
+
+//go:embed migrations/*.sql
+var migrations embed.FS
+
+var ErrEmptyDB = errors.New("DB is empty")
+
+const (
+	ledgerEntriesTableName      = "ledger_entries"
+	ledgerEntriesMetaTableName  = "ledger_entries_meta"
+	latestLedgerSequenceMetaKey = "LatestLedgerSequence"
+)
+
+type DB interface {
+	LedgerEntryStorage
+	GetLatestLedgerSequence() (uint32, error)
+	NewLedgerEntryUpdaterTx(forLedgerSequence uint32, maxBatchSize int) (LedgerEntryUpdaterTx, error)
+}
+
+type LedgerEntryUpdaterTx interface {
+	UpsertLedgerEntry(key xdr.LedgerKey, entry xdr.LedgerEntry) error
+	DeleteLedgerEntry(key xdr.LedgerKey) error
+	Done() error
+}
+
+type sqlDB struct {
+	db *sqlx.DB
+}
+
+func OpenSQLiteDB(dbFilePath string) (DB, error) {
+	db, err := sqlx.Open("sqlite3", dbFilePath)
+	if err != nil {
+		return nil, errors.Wrap(err, "open failed")
+	}
+
+	ret := &sqlDB{
+		db: db,
+	}
+
+	err = runMigrations(ret.db.DB, "sqlite3")
+
+	return ret, err
```

Comment:
I'd suggest that we'll be more atomic; if err != nil, do not return ret. ( and close the db )